server/src/main/java/io/crate/execution/ddl/tables/AlterTableRequest.java

```diff
@@ -73,7 +63,22 @@ public AlterTableRequest(StreamInput in) throws IOException {
         isPartitioned = in.readBoolean();
         excludePartitions = in.readBoolean();
         settings = readSettingsFromStream(in);
-        mappingDelta = in.readOptionalString();
+        if (in.getVersion().before(Version.V_5_10_0)) {
+            in.readOptionalString(); // mappingDelta
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        tableIdent.writeTo(out);
+        out.writeOptionalString(partitionIndexName);
+        out.writeBoolean(isPartitioned);
+        out.writeBoolean(excludePartitions);
+        writeSettingsToStream(out, settings);
+        if (out.getVersion().before(Version.V_5_10_0)) {
+            out.writeOptionalString(null);
+        }
```

Comment:
BWC related question:

on 5.10 node creating a table, column_policy will be passed via `writeSettingsToStream` and can be processed by a 5.10 node.

But what happens if a 5.10 node is a handler node, creates a table with non-default policy. 

Then a 5.9 node processes this request (because it's a master or say cluster went down during planning or we had re-election or whatever). 5.9 node will create a table without a policy (because it's streamed as null). 

It is a valid scenario?

I think we should keep writing delta along with including column_policy to settings and remove it later, say in 6.0?

 Another question (if all above is valid):
 Can AlterTableClusterStateExecutor on 5.9 handle new setting? If not we probably need to no only keep writing mappingDelta but also exclude column_policy from **writeSettingsToStream** for <5.10 nodes.

Reply:
Hm yes, good point. I wonder if we should stimply disallow running ALTER TABLE in a before 5.10 mix - because I might need make some more changes in this area and then things could get even more complicated.

Reply:
I think it's fine to add this protection, if we benefit of not adding bwc code. We did something similar for altering roles: https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/role/TransportAlterRoleAction.java#L92

Reply:
We agreed to allow DDL only on adjacent versions but as @matriv said, it's fine to forbid ALTER TABLE in a mixed cluster at all, especially if it allows to keep things simpler.

Reply:
>  We did something similar for altering roles: https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/role/TransportAlterRoleAction.java#L92

Added something like it - kept the BWC handling in the request to avoid getting hard to debug serialization error and ensure the check in the transport action master operation can actually run